astroquery/esasky/core.py

```diff
@@ -0,0 +1,624 @@
+        self._store_query_result_catalogs(query_result, sanatized_catalogs, coordinates, sanatized_radius, get_query_payload, cache)
+        
+        if(get_query_payload):
+            return query_result
+        
+        return commons.TableList(query_result)
+    
+    def get_maps(self, query_table_list, missions = __ALL_STRING, download_folder = __MAPS_STRING, cache = True):
+        """
+        This method takes the dictionary of missions and metadata as returned by query_region_maps
+        and downloads all maps to the selected folder.
+        The method returns a dictionary which is divided by mission and observation id. 
+
+        Parameters
+        ----------
+        query_table_list : `astroquery.utils.TableList`
+            A TableList with all the missions wanted and their respective metadata.
+            Usually the return value of query_region_maps.
+        missions : string or list, optional
+            Can be either a specific mission or a list of missions (all mission names are found in list_missions()) 
+            or 'all' to search in all missions. Defaults to 'all'. 
+        download_folder : string, optional
+            The folder where all downloaded maps should be stored. 
+            Defaults to a folder called 'Maps' in the current working directory. 
+
+        Returns
+        -------
+        maps : dict
+            Each mission returns dict where each observation id has a HDUList. 
+            The exception is Herschel as it has multiple fits files per observation id.
+            For Herschel each observation id got the used filters with their respective HDULists.
+            It is structured in a dictionary like this:
+                dict: {'XMM-EPIC': {'observationID_x': [HDUList], 'observationID_y': [HDUList], ...}, 'HST': {'observationID_z': [HDUList], ...},
+                    'HERSCHEL': {'observationID_a': {'filter_a':[HDUList], filter_b: [HDUList]}, 'observationID_b': {'filter_a':[HDUList], filter_b: [HDUList]}, ... }}
+        
+        Examples
+        --------
+        get_maps(query_region_catalogs("m101", "14'", "all"))
+        
+        """    
+        sanatized_query_table_list = self._sanatize_input_table_list(query_table_list)
+        sanatized_missions = self._sanatize_input_mission(missions)
+
+        maps = dict()
+         
+        for query_mission in sanatized_query_table_list.keys():
+            for mission in sanatized_missions:
+                #INTEGRAL does not have a product url yet.
+                if(query_mission.lower() == self.__INTEGRAL_STRING):
+                    break
+                if(query_mission.lower() == mission.lower()):
+                    maps[query_mission] = self._get_maps_for_mission(sanatized_query_table_list[query_mission], query_mission, download_folder, cache)
+                    break
+        if(len(sanatized_query_table_list) > 0):
+            print("Maps available at %s" %os.path.abspath(download_folder))
```

Comment:
Use `log.info` instead of `print` here (it gives users finer-tuned control over the code's verbosity)